sdk/tests/integration/TransferTransactionIntegrationTest.cc

```diff
@@ -0,0 +1,148 @@
+    {
+      nodeAddressString = jsonData[networkTag][accountIdStr];
+    }
+
+    if (jsonData[operatorTag][accountIdTag].is_string() && jsonData[operatorTag][privateKeyTag].is_string())
+    {
+      std::string operatorAccountIdStr = jsonData[operatorTag][accountIdTag];
+
+      operatorAccountId = AccountId::fromString(operatorAccountIdStr);
+      operatorAccountPrivateKey = jsonData[operatorTag][privateKeyTag];
+    }
+
+    testInputFile.close();
+
+    std::unordered_map<std::string, AccountId> networkMap;
+    networkMap.insert(std::pair<std::string, AccountId>(nodeAddressString, accountId));
+
+    mClient = Client::forNetwork(networkMap);
+    mClient.setOperator(operatorAccountId, ED25519PrivateKey::fromString(operatorAccountPrivateKey));
+  }
+
+private:
+  const std::string_view mJsonNetworkTag = "network";
+  const std::string_view mJsonOperatorTag = "operator";
+  const std::string_view mJsonAccountIdTag = "accountId";
+  const std::string_view mJsonPrivateKeyTag = "privateKey";
+
+  const std::string_view mNodeAccountIdStr = "0.0.3";
+  const AccountId mNodeAccountId = AccountId::fromString("0.0.3");
+  const AccountId mTestAccountId = AccountId::fromString("0.0.1023");
+  const Hbar mTestBalance = Hbar(10ULL, HbarUnit::TINYBAR());
+
+  const std::filesystem::path mFilePath = std::filesystem::current_path() / "local_node.json";
+
+  Client mClient;
+};
+
+// Tests invoking of method execute() from TransferTransaction.
+TEST_F(TransferTransactionIntegrationTest, ExecuteRequestToTestnetNode)
+{
+  // Given
+  const auto testBalance = getTestBalance();
+  const auto testPublicKey = ED25519PrivateKey::generatePrivateKey()->getPublicKey();
+  const auto accountOperatorId = AccountId::fromString("0.0.1024");
+  const auto testMemo = "Test memo for TransferTransaction.";
+
+  AccountCreateTransaction testAccountCreateTransaction;
+  testAccountCreateTransaction.setKey(testPublicKey);
+  testAccountCreateTransaction.setInitialBalance(testBalance);
+
+  TransactionReceipt testTxReceipt =
+    testAccountCreateTransaction.execute(getTestClient()).getRecord(getTestClient()).getReceipt().value();
+  AccountId createdAccountId = testTxReceipt.getAccountId().value();
+
+  TransferTransaction transferTransaction;
+  transferTransaction.addApprovedHbarTransfer(createdAccountId, Hbar(20ULL, HbarUnit::TINYBAR()));
+  transferTransaction.addApprovedHbarTransfer(accountOperatorId, Hbar(-20ULL, HbarUnit::TINYBAR()));
+  transferTransaction.setTransactionMemo(testMemo);
+
+  // When
+  const TransactionResponse txResponse = transferTransaction.execute(getTestClient());
+
+  // Then
+  const TransactionRecord txRecord = txResponse.getRecord(getTestClient());
+  const TransactionReceipt txReceipt = txResponse.getReceipt(getTestClient());
+  EXPECT_TRUE(txResponse.getValidateStatus());
+  EXPECT_TRUE(txRecord.getReceipt().has_value());
+  EXPECT_TRUE(txRecord.getConsensusTimestamp().has_value());
+  EXPECT_EQ(txRecord.getTransactionMemo(), testMemo);
```

Comment:
There are fields in `TransactionRecord` where you can see all the individual transfers that occurred as part of the transaction. Those should probably be looked at and verified here.

Reply:
Thanks for noticing it! 
I used `TransferTransactionTest.cc` as an example how to setup properly an instance of `TransactionRecord`.
Fixed version is introduced in 96d804517da2be796274d53a3648a5f6e93fbe8c.